frontend/src/models/models.js

```diff
@@ -89,6 +90,41 @@ module.exports = app => app.component('models', {
     this.status = 'loaded';
   },
   methods: {
+    clickFilter(path) {
+       // Initialize searchFieldsMap if empty
+      if (!this.searchText) {
+        this.searchFieldsMap = {};
+      }
+
+      // Extract existing key-value pairs from searchText
+      const regex = /(\w+):\s*([^,}]*)/g;
+      let match;
+      while ((match = regex.exec(this.searchText)) !== null) {
+        const key = match[1].trim();
+        const value = match[2].trim();
+        this.searchFieldsMap[key] = value || ''; // Preserve values or default to empty
+      }
+
+      // Toggle the clicked property
+      if (this.searchFieldsMap.hasOwnProperty(path)) {
+        delete this.searchFieldsMap[path]; // Remove if already exists
+      } else {
+        this.searchFieldsMap[path] = ''; // Add new property with empty value
+      }
+
+      // Reconstruct searchText with preserved values
+      this.searchText = `{ ${Object.entries(this.searchFieldsMap)
```

Comment:
Overall, I'd prefer an approach that doesn't affect the user's existing input because modifying existing input is risky. Maybe we just add `, userId: ` right before the ending `}` if the input ends with `}`?